tests/integration/systems/test_systems.py

```diff
@@ -0,0 +1,156 @@
+    ):
+        create_virtual_system_request: CreateVirtualSystemRequest = (
+            CreateVirtualSystemRequest(
+                alias="Python integration virtual system",
+                workspace=workspace_id,
+            )
+        )
+
+        create_virtual_system_response = create_virtual_systems(
+            create_virtual_system_request
+        )
+
+        minion_id = (
+            create_virtual_system_response.minionId
+            if create_virtual_system_response
+            and create_virtual_system_response.minionId
+            else None
+        )
+
+        assert minion_id is not None
+
+    def test__query_systems__returns_queried_virtual_system(
+        self, client: SystemsClient, create_virtual_systems
+    ):
+        create_virtual_system_request: CreateVirtualSystemRequest = (
+            CreateVirtualSystemRequest(
+                alias="Python integration virtual system",
+                workspace=workspace_id,
+            )
+        )
+
+        create_virtual_system_response = create_virtual_systems(
+            create_virtual_system_request
+        )
+
+        minion_id = (
+            create_virtual_system_response.minionId
+            if create_virtual_system_response
+            and create_virtual_system_response.minionId
+            else None
+        )
+
+        assert minion_id is not None
+
+        query_virtual_system_response: QuerySystemsResponse = client.query_systems(
+            query=QuerySystemsRequest(filter=f'id="{minion_id}"')
+        )
+
+        assert query_virtual_system_response.data is not None
+        assert query_virtual_system_response.data[0]["id"] == minion_id
+
+    def test__remove_sytems(self, client: SystemsClient):
+
+        create_virtual_system_request: CreateVirtualSystemRequest = (
+            CreateVirtualSystemRequest(
+                alias="Python integration query virtual system",
+                workspace=workspace_id,
+            )
+        )
+
+        create_virtual_system_response: CreateVirtualSystemResponse = (
+            client.create_virtual_system(
+                create_virtual_system_request=create_virtual_system_request
+            )
+        )
+
+        minion_id = (
+            create_virtual_system_response.minionId
+            if create_virtual_system_response
+            and create_virtual_system_response.minionId
+            else None
+        )
+
+        assert minion_id is not None
+
+        remove_system_response: RemoveSystemsResponse = client.remove_systems(
+            tgt=[minion_id]
+        )
+
+        assert remove_system_response.removed_ids is not None
+        assert remove_system_response.removed_ids[0] == minion_id
+
+    def test__query_systems_with_projections__returns_the_systems_with_projected_properties(
+        self, client: SystemsClient
+    ):
+        query_systems_request = QuerySystemsRequest(
+            projection="new(id, alias, grains.data.master, grains.data.host)", take=1
```

Comment:
```suggestion
            projection="new(id, alias, grains.data.master as master, grains.data.host as host)", take=1
```
This is more representative of what we'd want callers to do

Reply:
Updated as per suggestion.